### DISCUSSION TITLE
When using getStaticProps, does it matter for loading speed if heavy logic is executed inside getStaticProps or the component itself?

### DISCUSSION THREAD

**Discussion by florianwalther-private** [vercel/next.js#39180] [Help]
My blog posts are all pre-rendered using `getStaticProps`. Right now, I'm parsing markdown to HTML inside `getStaticProps` and do some processing there.
I now want to switch to [react-markdown](https://github.com/remarkjs/react-markdown), which moves the markdown processing into the component itself. Will this have an impact on the loading speed of the page, or does it not matter because it's all run and compile-time?

**Comment by GabenGar** [comment]
That entirely depends on how `react-markdown` works. Do you see the parsed markdown in initial html?

**Reply by florianwalther-private** [reply]
I might understand it wrong, but I was under the impression that getStaticProps renders the whole page to HTML at compile time and doesn't require any JavaScript at runtime at all. So, I thought react-markdown would run at compile-time, not when the user actually opens the page. Is that wrong?

> Do you see the parsed markdown in initial html?

What do you mean exactly?

**Reply by GabenGar** [reply]
>I might understand it wrong, but I was under the impression that getStaticProps renders the whole page to HTML at compile time and doesn't require any JavaScript at runtime at all. 

I am not talking about `getStaticProps()`, I am talking about `react-markdown`. Chances are it takes a markdown file and parses it at runtime. Also "doesn't require any JavaScript at runtime at all" heavily depends on the resulting react markup.

>What do you mean exactly?

The HTML file you get on request before it gets hydrated.

**Reply by florianwalther-private** [reply]
> The HTML file you get on request before it gets hydrated.

I have no idea, to be honest. But I need react-markdown's features so I think I'll just accept whatever baggage it brings (or not).

**Reply by sillvva** [reply]
`react-markdown` processes the files at compile-time when using `getStaticProps`.

**Reply by florianwalther-private** [reply]
> `react-markdown` processes the files at compile-time when using `getStaticProps`.

Ok that's great news then!
